examples/cert-validation.cpp

```diff
@@ -0,0 +1,248 @@
+        ctx.addTrustedCertificate(rootCA)
+           .addIntermediateCertificate(intermediateCA)
+           .enforceSelfSignedRootCertificate();
+        std::cout << "Checking if root certificate is self signed\n";
+        try {
+            cert.verify(ctx);
+        } catch (const MoCOCrWException &e) {
+            std::cerr << "Root certificate probably not self signed.\n";
+            std::cerr << "OpenSSL error: " << e.what() << '\n';
+            exit(EXIT_FAILURE);
+        }
+        std::cout << "Root certificate is self signed\n---\n";
+    }
+
+    /*
+        Enforce self signed root certificate in VerificationContext using
+        enforceSelfSignedRootCertificate(). In this case intermediate CA
+        certificate is not self-signed so exception is expected.
+    */
+    {
+        X509Certificate::VerificationContext ctx;
+        ctx.addTrustedCertificate(intermediateCA)
+           .enforceSelfSignedRootCertificate();
+        std::cout << "Checking if intermediate certificate is self signed\n";
+        try {
+            rootCA.verify(ctx);
+        } catch (const MoCOCrWException &e) {
+            std::cerr << "Failed as expected.\n";
+            std::cerr << "OpenSSL error: " << e.what() << '\n';
+        }
+        std::cout << "---\n";
+    }
+
+    /*
+        Add CRL to VerificationContext for intermediate CA which has
+        ceritificate stored in variable `cert` revoked (serial number 0x1002).
+        CRL's are added with addCertificateRevocationList() method.
+        Exception is thrown because validated certificate is a part of
+        root CA's CRL.
+    */
+    {
+        X509Certificate::VerificationContext ctx;
+        ctx.addTrustedCertificate(rootCA)
+           .addIntermediateCertificate(intermediateCA)
+           .addCertificateRevocationList(intermediateCRL);
+        std::cout << "Checking if certificate is revoked\n";
+        try {
+            cert.verify(ctx);
+        } catch (const MoCOCrWException &e) {
+            std::cout << "Failed as expected.\n";
+            std::cerr << "OpenSSL error: " << e.what() << '\n';
+        }
+        std::cout << "---\n";
+    }
+
+    /*
+        Intermediate CA is missing CRL but enforceCrlsForAllCAs is set.
+        Exception is expected.
+    */
+    {
+        X509Certificate::VerificationContext ctx;
+        ctx.addTrustedCertificate(rootCA)
+           .addIntermediateCertificate(intermediateCA)
+           .addCertificateRevocationList(rootCRL)
+           .enforceCrlsForAllCAs()
+           .enforceSelfSignedRootCertificate();
+        std::cout << "Checking enforcement of CRL's\n";
+        try {
+            cert.verify(ctx);
+        } catch (const MoCOCrWException &e) {
+            std::cerr << "Failed as expected.\n";
+            std::cerr << "OpenSSL error: " << e.what() << '\n';
+        }
+        std::cout << "---\n";
+    }
+
+    /*
+        Adding custom time in VerificationContext
```

Comment:
Maybe be a bit more verbose here. Something like. 

> The VerificationContext also allows the caller to specify a specific time (and date) that should be used for certificate validation instead of the current system time.